client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -158,25 +214,58 @@ tie.factory('PythonCodeRunnerService', [
       );
     };
 
-    var _processCodeExecutionServerResponse = function(responseData, code) {
+    var _processCodeExecutionServerResponse = function(
+        responseData, code, separator) {
       if (responseData.stderr.length) {
         var errorTraceback = ErrorTracebackObjectFactory.fromPythonError(
           responseData.stderr);
         return CodeEvalResultObjectFactory.create(
             code, '', [], [], [], errorTraceback,
             responseData[VARNAME_MOST_RECENT_INPUT]);
       } else if (responseData.results) {
-        return CodeEvalResultObjectFactory.create(
-            code, responseData.stdout,
+        var standardizedOutput = _standardizeServerOutput(
+          responseData.stdout, separator);
+        var codeEvalResult = CodeEvalResultObjectFactory.create(
+            code, standardizedOutput,
             responseData.results[VARNAME_OBSERVED_OUTPUTS],
             responseData.results[VARNAME_BUGGY_OUTPUT_TEST_RESULTS],
             responseData.results[VARNAME_PERFORMANCE_TEST_RESULTS],
             null, null);
+        _displayPrintedOutput(codeEvalResult);
+        return codeEvalResult;
       } else {
         throw Error('A server error occurred. Please try again.');
       }
     };
 
+    // Converts the output string generated by running the code.
+    // Before: string of all stdOut concatenated together.
+    // After: array in which the i-th entry is the string corresponding
+    // to the stdOut of the i-th test case.
+    var _standardizeServerOutput = function(stdOutString, separator) {
```

Comment:
As above -- don't have separate functions for client and server. Instead, update the stdout after runCode so that it is in a consistent format in both cases, and so that there's no need for future execution to know whether the client or the server was used for execution.